Assets/Scripts/UserInterface/UIButton.cs

```diff
@@ -0,0 +1,70 @@
+/**
+ * Author(s): Ranjini Das
+ **/
+
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+
+public class UIButton : MannBehavior
+{
+    GameObject gameObject;
+    EventController.Action action; 
+    bool isDown;
+    public delegate void performAction (EventController.Action action);  
+
+    public UIButton (GameObject gameObject, EventController.Action action)
```

Comment:
Definitely on the right track. Note that the MonoBehaviour superclass will set the gameObject reference for you.

Also, would refer to the note above about subscribing actions in order to call multiple actions on button press. So the EventController.Action argument looks good, but you'll likely want to subscribe it to an event instead of just setting a direct reference (this pattern could also handle receiving an array of events, and then just loop through and subscribe all of them).